jsapp/js/components/permissions/utils.ts

```diff
@@ -460,6 +461,66 @@ export function getPermLabel(perm: PermissionResponse) {
   return '???'
 }
 
+/**
+ * Returns the permission label with the correct suffix depending on the given asset type.
+ *
+ * Example: if we are sharing a library collection, the permissions will all say "[View, Edit, Manage] collection"
+ */
+export function getFriendlyPermLabelWithSuffix(
+  assetType: AssetTypeName | undefined,
+  perm?: PermissionResponse,
+  checkboxNameParam?: CheckboxNameAll,
+) {
+  let checkboxNameSuffix = ''
+  let checkboxNameAll = ''
+
+  if (perm) {
+    const permDef = permConfig.getPermission(perm.permission)
+
+    if (permDef) {
+      checkboxNameAll = getCheckboxNameByPermission(permDef.codename) || ''
+    }
+  }
+
+  if (checkboxNameParam) {
+    checkboxNameAll = checkboxNameParam
+  }
+
+  if (checkboxNameAll === 'formView' || checkboxNameAll === 'formEdit' || checkboxNameAll === 'formManage') {
+    switch (assetType) {
+      case AssetTypeName.block:
+        if (checkboxNameAll === 'formView') checkboxNameSuffix = t('View block')
+        if (checkboxNameAll === 'formEdit') checkboxNameSuffix = t('Edit block')
+        if (checkboxNameAll === 'formManage') checkboxNameSuffix = t('Manage block')
+        break
```

Comment:
If you would use `return` in each line, you don't need `break`:

```ts
        if (checkboxNameAll === 'formManage') return t('Manage block')
```

Reply:
And then we don't need `checkboxNameSuffix` variable, and we can rename `checkboxNameAll` (confusing) to `checkboxName`

Reply:
I like doing it this way because it removes the confusing `...Suffix` variable, but adding return to each line still gets typescript to complain about fallthrough. In order to not have a `break` be necessary the code would look like this:
```typescript
        if (checkboxName === 'formView') return t('View block')
        else if (checkboxName === 'formEdit') return t('Edit block')
        else return t('Manage block')
```

which I think is worse than a `break` at the end, let me know what you think :)

Reply:
The compromise I made: https://github.com/kobotoolbox/kpi/pull/5736/commits/64268fa3de1f10cc0658bbd77063426c8fbc6ca1